src/test/java/net/sf/jabref/importer/fileformat/BibtexParserTest.java

```diff
@@ -1486,4 +1487,43 @@ public void parseReturnsEntriesInSameOrder() throws IOException {
 
         assertEquals(expected, result.getDatabase().getEntries());
     }
+
+    @Test
+    public void parsePrecedingComment() throws IOException {
+        // @formatter:off
+        String bibtexEntry = "% Some random comment that should stay here" + Globals.NEWLINE +
+                "@Article{test," + Globals.NEWLINE +
```

Comment:
Add test for entries starting on the same line as the comment, i.e. `some text containing % and now the @article {...}`